File: practica2_820574_839304/c++/swap.cpp
#include "swap.h"

Swap::Swap() {
    nombre = "swap";
}

void Swap::ejecutar(PilaInts &pila, int &pc) {
    int a, b;
    a = pila.cima();
    pila.extraer();
    b = pila.cima();
    pila.extraer();

    pila.insertar(b);
    pila.insertar(a);
    pc++;
}

string Swap::nombrar() {
    return nombre;
}

File: practica2_820574_839304/java/Swap.java
/*
* Fichero: Swap.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Stack;

public class Swap extends Instruccion{
    Swap(){
        nombre = "swap";
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        int a, b;
        a = pila.pop();
        b = pila.pop();
        pila.push(a);
        pila.push(b);
        pc[0]++;
    }
}